otter/util/logging_treq.py

```diff
@@ -24,16 +24,23 @@ def _log_request(treq_call, url, **kwargs):
     """
     clock = kwargs.pop('clock', reactor)
     log = kwargs.pop('log', None)
+
     if not log:
         log = default_log
     method = kwargs.get('method', treq_call.__name__)
 
-    treq_transaction = str(uuid4())
+    kwargs.setdefault('headers', {})
+    treq_transaction = kwargs['headers'].setdefault('x-otter-request-id',
+                                                    [str(uuid4())])[0]
```

Comment:
Sounds good - if we actually come up with a use case we can try to preserve it.  Will remove this and add in a test :D